### PR TITLE
Crypto account

### PR DESCRIPTION
**Pull request by vimalan-deriv**
In this testcase we have added checks to check on the deposit flow of all cryptocurrency to ensure that the address and QR code in deposit page are generated. 

Clickup card to refer: https://app.clickup.com/t/86bxa0c1w

### CODE REVIEW

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST-707 - Create crypto account', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:3]: Linked card links to QATEST-715. Please update the description. If required, you may merge both tests if they share functionality.

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        .should('be.visible')
+        .click()
+      cy.findByText(crypto).click()
+      cy.findByTestId('dt_currency-switcher__arrow').should('be.visible')
+      cy.c_closeNotificationHeader()
+      cy.findByRole('button', { name: 'Deposit' }).click()
+
+      // To check for QR code and compare with the code generated
+      cy.xpath('//canvas[contains(@class,"qrcode")]').should('be.visible') // assert that the QR code element exists
+      cy.xpath('//canvas[contains(@class,"qrcode")]').then(($canvas) => {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:31]: Can you please use `cy.get("canvas[class*='qrcode']")` instead? XPath usage is not recommended and must be an exception.

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                )
+              })
+            })
+        })
+      cy.findAllByText('Transaction status').should('be.visible')
+      cy.get('.transactions-crypto-transaction-status-side-note').should(
+        'exist'
+      )
+      cy.get('.side-note').should('exist')
+      cy.findAllByTestId('dt_traders_hub_home_button').click()
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:82]: Does this work? If it's finding multiple elements, please specify by index which element you want to click. If not, please use findByTestId. OR, use click({multiple:true}) if you're trying to click multiple elements.

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import '@testing-library/cypress/add-commands'
+
+describe('QATEST-707 - Create crypto account', () => {
+  beforeEach(() => {
+    cy.c_login({ user: 'allcrypto' })
+    cy.c_visitResponsive('/appstore/traders-hub', 'large')
+  })
+
+  it('should be able to create crypto account from Traders Hub.', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:9]: Same as above for this.

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
Left some comments related to test description and locators.

**Review by abuzar-deriv — APPROVED** [review]
LGTM.

File: cypress.config.js [hunk 5]
```diff
@@ -76,6 +76,10 @@ module.exports = defineConfig({
         ID: process.env.E2E_P2P_FLOATING,
         PSWD: process.env.E2E_QA_ACCOUNT_PASSWORD
       },
+      allcrypto: {
+        ID: process.env.E2E_CRYPTO,
+        PSWD: process.env.E2E_CRYPTO_PASSWORD,
```
Comment (meeyun-deriv) [inline cypress.config.js:81]: can replace this with `E2E_QA_ACCOUNT_PASSWORD`

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+          imageData.width,
+          imageData.height
+        )
+        // Assert that the QR code was successfully decoded and get its value
+        expect(code).to.not.be.null
+        const qrCodeValue = code.data
+        // Log the value of the QR code
+        cy.log('QR Code Value:', qrCodeValue)
+        // Get the value from the element with the specified class
+        cy.get('.deposit-crypto-wallet-address__hash-container')
```
Comment (meeyun-deriv) [inline cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:48]: quite some elements are using classname, did we create a card to let dev to update the locators with best practice?

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]

File: cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js (hunk 6 above)
Comment (meeyun-deriv) [reply cypress/e2e/wip/Crypto/crypto_deposit_check.cy.js:48]: locator will get update in this card https://app.clickup.com/t/86byjmgdg

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Pull request merged**